advisor-backend/routes/events.js

```diff
@@ -1,11 +1,26 @@
 import express from "express";
 import { EventDetails } from "../models/event.js";
-
+// number of elements to be displayed on the page
+const pageLimit = 8;
 const router = express.Router();
 
-router.get("/", async (req, res) => {
+router.get("/page-count", async (req, res) => {
+  try {
+    const count = await EventDetails.count();
+    const pageCount = Math.ceil(count / pageLimit);
+    res.status(200).json({ pageCount });
+  } catch (error) {
+    res.status(500).json({ error: "Server error" });
+  }
+});
+
+router.get("/:page", async (req, res) => {
+  const page = req.params.page;
   try {
-    const events = await EventDetails.findAll();
+    const events = await EventDetails.findAll({
```

Comment:
One thing to double check here is if `findAll` is always going to use the same order (e.g. does it iterate by primary key or something). 

I imagine it tries to, but there might be cases where it won't in which case you'd want to add an `orderBy` or something equivalent.